In a weird coincidence, @gottesmm added a test checking for the lack of diagnostic since I branched from master:

```
20:32:59 /Users/buildnode/jenkins/workspace/swift-PR-osx/branch-master/swift/test/Interpreter/bridged_casts_folding.swift:309:12: error: CHECK: expected string not found in input
20:32:59  // CHECK: stderr>>> OK: saw expected "crashed: sigill"
20:32:59            ^
20:32:59 <stdin>:34:1: note: scanning from here
20:32:59 stderr>>> Could not cast value of type 'NSTaggedPointerString' (0x7fffa43976b0) to 'NSNumber' (0x7fffa4848ac0).
20:32:59 ^
20:32:59 <stdin>:35:1: note: possible intended match here
20:32:59 stderr>>> OK: saw expected "crashed: sigabrt"
```

The failure in `IRGen/class_resilience.swift` is unrelated.

--- a/stdlib/public/runtime/Casting.cpp
+++ b/stdlib/public/runtime/Casting.cpp
@@ -3023,14 +3023,17 @@ _bridgeNonVerbatimFromObjectiveC(
         const_cast<void*>(swift_dynamicCastUnknownClass(sourceValue,
                                                         objectiveCType));
       
-    if (sourceValueAsObjectiveCType) {
-      // The type matches.  _forceBridgeFromObjectiveC returns `Self`, so
-      // we can just return it directly.
-      bridgeWitness->forceBridgeFromObjectiveC(
-        static_cast<HeapObject*>(sourceValueAsObjectiveCType),
-        destValue, nativeType, nativeType, bridgeWitness);
-      return;
+    if (!sourceValueAsObjectiveCType) {
+      swift::swift_dynamicCastFailure(_swift_getClass(sourceValue),
+                                      objectiveCType);
     }
+
+    // The type matches.  _forceBridgeFromObjectiveC returns `Self`, so
+    // we can just return it directly.
+    bridgeWitness->forceBridgeFromObjectiveC(
+      static_cast<HeapObject*>(sourceValueAsObjectiveCType),
+      destValue, nativeType, nativeType, bridgeWitness);
+    return;
   }
   
   // Fail.

